### PR TITLE
Get monthly fee + Get 3 months fee (this month + prev 2 months) function (Only supports non-recurring session for now)

### PR DESCRIPTION
**Pull request by enhao25**
Update with the following feature:
1. Get monthly fee (E.g. ` fee n/John Doe m/1 y/2021`) - Returns monthly fee of a particular student on the specific month and year
2. Get 3 months fee (this month + prev 2 months) (E.g 3Month_Fee) - Returns past 3 months fee based on current list

Considerations:
1. Only allows month between 1 to 12
2. Only allows year between 1970 to 2037

Future Issues / Improvements:
1. Test cases for command and command parser
2. Might not need the command for 3Month_Fee if this is automatically in the UI.
3. ~~Only works on the filtered student, find_student will change the value (Could make it for all students instead of filtered students)~~
4. Take into consideration for recurring session when it is done.

#81 #82

### CODE REVIEW

**Review by JonahhGohh — APPROVED** [review]
Tested it and works well!
I like how you compared the dateTime with the compareTo() method, clean af haha.

Regarding the 3 month command, I was thinking of using it for the Home fee section. So not too sure if we need a command for it.

**Comment by enhao25** [comment]
> Tested it and works well!
> I like how you compared the dateTime with the compareTo() method, clean af haha.
> 
> Regarding the 3 month command, I was thinking of using it for the Home fee section. So not too sure if we need a command for it.

Haha I agree ah! For the 3 month command thingy, I was also thinking that we might not need the command. So haven't added it to the UG yet. Probably can keep the command for testing purposes, and remove it after the UI that uses the 3 month function is done.

File: src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                    // This session date is within the period
+                    fee += session.getFee().getFee();
+                }
+            }
+        }
+        return fee;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```
Comment (nowknowing) [inline src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java:32]: Im understanding that getPrev3 month is getMonth applied to every student, and for 3 months. I guess it'd be considered less "repeated code" if this method was built on getMonth. possibly using getMonth's methods from within getPrev3?

**Review by nowknowing — COMMENTED** [review]
I think it works well! Just maybe could reduce repeat but i can't remember if we're marked on that. If not I think this is good!

File: src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java (hunk 1 above)
Comment (enhao25) [reply src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java:32]: Ok good suggestion, the only common place for both code base is on model. Hence, I have abstracted out the get fees code over to model. Do have a look, the output should still be the same. Thank you!

File: src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        return LocalDateTime.of(year.getYear(), month.getMonth(), 1, 0, 0);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        LocalDateTime currMonthYear;
+        LocalDateTime nextMonthYear;
+        StringBuilder resultString = new StringBuilder("Total Monthly fee for past 3 months is:");
```
Comment (nowknowing) [inline src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java:37]: Could possibly remove "total", just let it be "Monthly fee for the past 3 months"

File: src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.fee.Month;
+import seedu.address.model.fee.Year;
+
+/**
+ * Gets current month and previous 2 months expected monthly fee based on the current list of sessions
+ */
+public class GetPrev3MonthFeeCommand extends Command {
+    public static final String COMMAND_WORD = "3Month_Fee";
```
Comment (nowknowing) [inline src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java]: Might be consistent with the other commands if not capitalised. eg "past_fees" / "3_fees" / "3_month_fees"

**Review by nowknowing — APPROVED** [review]
LGTM!!

File: src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Shows the total expected monthly fee"
+        + "for this month + previous 2 months";
```
Comment (nowknowing) [inline src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java:20]: "Shows the totalled fees per month, for the past 3 months." Or something of the sort could be somewhat more accurate.

File: src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java (hunk 1 above)
Comment (nowknowing) [reply src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java:32]: super neat, hadn't thought of putting at model!

Comment (nowknowing) [reply src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java:32]: super neat, hadn't thought of putting it at model!

File: src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java (hunk 3 above)
Comment (enhao25) [reply src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java]: Ahhh right! I have changed it to 3_month_fees instead.

File: src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java (hunk 4 above)
Comment (enhao25) [reply src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java:20]: Alright! Change from that to "Shows the totalled fees per month, for this month + previous 2 months" instead.

**Comment by enhao25** [comment]
closes #81

**Pull request merged**